### PR TITLE
CV2-5608: remove show_parent? method

### PR DESCRIPTION
**Pull request by melsawy**
## Description

Simplify and get rid of the show_parent? logic so search all items and either show parent or secondary items based on `show_similar` option
i.e show_similar = false then show parent items only otherwise show secondary items

References: CV2-5608

## How has this been tested?

Implemented automated tests.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

**Review by jayjay-w — APPROVED** [review]

**Review by caiosba — CHANGES_REQUESTED** [review]
@melsawy , please add a proper PR description

**Comment by melsawy** [comment]
Sorry @caiosba it was a draft and I thought I added a description so I just click ready for review.

File: lib/check_search.rb [hunk 1]
```diff
@@ -241,7 +241,7 @@ def get_pg_results_for_media
       custom_conditions[k] = [@options[v]].flatten if @options.has_key?(v)
     end
     core_conditions.merge!({ archived: @options['archived'] })
-    core_conditions.merge!({ sources_count: 0 }) unless should_include_related_items?
+    core_conditions.merge!({ sources_count: 0 }) unless @options['show_similar']
```
Comment (caiosba) [inline lib/check_search.rb:244]: @melsawy why we need to keep this logic for PG-only queries? I saw you removed it for ES queries, as expected.

Comment (melsawy) [reply lib/check_search.rb:244]: For the PG (cases not hit ES) the `sources_count`  is the way we can know if this is a parent or child but ES we have a `parent_id` field 
so the cases for search by keyword, status, tags, etc we already hit ES so I think it's save to keep the condition for PG only

**Review by caiosba — APPROVED** [review]
Good to merge after adding the comment I asked for, and making sure tests pass.

**Pull request merged**